### PR TITLE
Fatal error if admin enters wrong values for some customer attribute and uploads file

### LINKED ISSUE #30295: Fatal error if admin enters wrong values for some customer attribute and uploads file
If Admin tries to save customer entering not valid values for some attribute which has Input Validation set up, and uploads file at the same time he gots fatal error.

**Please note:** if you just enter wrong value for some attribute XOR upload file you will not get fatal.

**Steps to Reproduce:**
1. Create new Customer attribute for URL (Stores -> Attributes -> Customer):
# * Input Type = Text Field
# * Values Required = Yes
# * Default Value = www.google.com
# * Input Validation = URL

1. Create new Customer attribute for file:
# * Input Type = File
# * Values Required = Yes

1. Create new customer:
# * URL field - leave default value
# * File attribute - upload some image

1. Fill all other required fields
1. Click Save button

**Actual Result:**
 (-) Fatal Error
 **Note:** if you click back button in browser or open some other page in backend you will see error messages about field which had wrong values

**Expected Result:**
 (/) Error message appears for each field which has wrong value

### PR DESCRIPTION
**Pull request by engcom-Kilo**
### Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/422

### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/30295

### Manual testing scenarios (*)

1. see https://github.com/magento/magento2/issues/30295

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

File: app/code/Magento/Customer/Controller/Adminhtml/Index/Save.php [hunk 1]
```diff
@@ -185,7 +186,7 @@ protected function _extractCustomerData()
         }
 
         if (isset($customerData['disable_auto_group_change'])) {
-            $customerData['disable_auto_group_change'] = (int) filter_var(
+            $customerData['disable_auto_group_change'] = (int)filter_var(
```
Comment (zakdma) [inline app/code/Magento/Customer/Controller/Adminhtml/Index/Save.php]: Why do you remove this space? Its ok according to PSR-12
https://www.php-fig.org/psr/psr-12/#61-unary-operators

File: app/code/Magento/Customer/Controller/Adminhtml/Index/Save.php [hunk 2]
```diff
@@ -353,15 +353,15 @@ public function execute()
...
+                        throw new LocalizedException(__("The Store View selected for sending Welcome email from" .
                             " is not related to the customer's associated website."));
                     }
                 }
 
                 // Save customer
                 if ($customerId) {
                     $this->_customerRepository->save($customer);
-
+                    // @phpstan-ignore-next-line
```
Comment (zakdma) [inline app/code/Magento/Customer/Controller/Adminhtml/Index/Save.php]: what is the reason phpstan ignore this line?

File: app/code/Magento/Customer/Test/Mftf/ActionGroup/AdminAssertCustomerErrorMessageActionGroup.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminAssertCustomerErrorMessageActionGroup">
```
Comment (zakdma) [inline app/code/Magento/Customer/Test/Mftf/ActionGroup/AdminAssertCustomerErrorMessageActionGroup.xml]: Please rename AG according to rules and make it more descriptive
Should be for ex. **AssertAdminCustomerEditFormErrorMessage**
Also it is bad idea to check only one type of messages, especially error type. This AG reusable is too small.
Better create universal message check AG that take type of messages as argument
See **AssertMessageInAdminPanelActionGroup** as an example. Try to reuse **AssertMessageInAdminPanelActionGroup** if possible (remove createdErrorMessage in this case as redundant)

File: app/code/Magento/Customer/Test/Mftf/ActionGroup/AdminAssertCustomerErrorMessageActionGroup.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        <annotations>
+            <description>Verify error message on admin customer page.</description>
+        </annotations>
+
+        <arguments>
+            <argument name="text" type="string"/>
```
Comment (zakdma) [inline app/code/Magento/Customer/Test/Mftf/ActionGroup/AdminAssertCustomerErrorMessageActionGroup.xml]: Please make argument name more descriptive. For ex. **messageText**

File: app/code/Magento/Customer/Test/Mftf/ActionGroup/AdminAssertCustomerErrorMessageActionGroup.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        </arguments>
+        <see selector="{{NewCustomerPageSection.createdErrorMessage}}" userInput="{{text}}"  stepKey="seeCustomerError"/>
```
Comment (zakdma) [inline app/code/Magento/Customer/Test/Mftf/ActionGroup/AdminAssertCustomerErrorMessageActionGroup.xml]: It is better to wait message before check it
<waitForElementVisible selector="{{NewCustomerPageSection.createdErrorMessage}}" stepKey="waitErrorMessageAppeared"/>

File: app/code/Magento/Customer/Test/Unit/Model/FileProcessorTest.php [hunk 6]
```diff
@@ -313,6 +313,14 @@ public function testMoveTemporaryFileUnableToCreateDirectory()
 
         $destinationPath = 'customer/f/i';
 
+        $this->mediaDirectory->expects($this->at(0))
```
Comment (zakdma) [inline app/code/Magento/Customer/Test/Unit/Model/FileProcessorTest.php]: Please extract this repeatable code to separate method

**Review by zakdma — CHANGES_REQUESTED** [review]

File: app/code/Magento/Customer/Controller/Adminhtml/Index/Save.php (hunk 1 above)
Comment (engcom-Kilo) [reply app/code/Magento/Customer/Controller/Adminhtml/Index/Save.php]: Reverted

File: app/code/Magento/Customer/Controller/Adminhtml/Index/Save.php (hunk 2 above)
Comment (engcom-Kilo) [reply app/code/Magento/Customer/Controller/Adminhtml/Index/Save.php]: Reverted

File: app/code/Magento/Customer/Test/Mftf/ActionGroup/AdminAssertCustomerErrorMessageActionGroup.xml (hunk 3 above)
Comment (engcom-Kilo) [reply app/code/Magento/Customer/Test/Mftf/ActionGroup/AdminAssertCustomerErrorMessageActionGroup.xml]: Action group deleted

File: app/code/Magento/Customer/Test/Mftf/ActionGroup/AdminAssertCustomerErrorMessageActionGroup.xml (hunk 4 above)
Comment (engcom-Kilo) [reply app/code/Magento/Customer/Test/Mftf/ActionGroup/AdminAssertCustomerErrorMessageActionGroup.xml]: Action group is deleted

File: app/code/Magento/Customer/Test/Mftf/ActionGroup/AdminAssertCustomerErrorMessageActionGroup.xml (hunk 5 above)
Comment (engcom-Kilo) [reply app/code/Magento/Customer/Test/Mftf/ActionGroup/AdminAssertCustomerErrorMessageActionGroup.xml]: Action group is deleted

File: app/code/Magento/Customer/Test/Unit/Model/FileProcessorTest.php (hunk 6 above)
Comment (engcom-Kilo) [reply app/code/Magento/Customer/Test/Unit/Model/FileProcessorTest.php]: Done

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Review by zakdma — APPROVED** [review]

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@engcom-Kilo the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-30963`.

**Comment by engcom-Bravo** [comment]
### Manual testing scenario
1. Go to _Stores - Attributes - Customer_ click **Add New Attribute**
2. Set the following values: 
- **Default Label** - _New Customer Attribute_
- **Attribute Code** - _newattcode_
- **Input Type** - _Text Field_
- **Values Required** - _Yes_
- **Default Value** - _www.google.com_
- **Input Validation** - _URL_
- **Show on Storefront** - _Yes_
- **Sort Order** - _1_
3. Click **Save Attribute**
4. Go **Back** and click **Add New Attribute** again
5. Set the following values:
- **Default Label** - _New Customer Attribute for file_
- **Attribute Code** - _newattcodefile_
- **Input Type** - _File (attachment)_
- **Values Required** - _Yes_
- **Show on Storefront** - _Yes_
- **Sort Order** - _2_
6. Go to _Customers - All Customers_ click **Add New Customer**
7. **Upload** an image under _New Attribute Code for file_ and fill in all the required fields. (Fname, Lname, Email)
8. Click **Save Customer**

**BEFORE** applying changes provided in this PR
Two Exceptions appeared
```
2 exception(s):
Exception #0 (Magento\Framework\Exception\LocalizedException): Something went wrong while saving the file.
Exception #1 (Magento\Framework\Exception\FileSystemException): The "/sites/mage24developee/pub/media/customer/tmp/adobe-base.jpg" file doesn't exist.

Exception #0 (Magento\Framework\Exception\LocalizedException): Something went wrong while saving the file.
<pre>#1 Magento\Customer\Model\Metadata\Form\File->processUiComponentValue() called at [app/code/Magento/Customer/Model/Metadata/Form/File.php:297]
#2 Magento\Customer\Model\Metadata\Form\File->compactValue() called at [app/code/Magento/Customer/Model/Metadata/Form.php:264]
#3 Magento\Customer\Model\Metadata\Form->compactData() called at [generated/code/Magento/Customer/Model/Metadata/Form/Interceptor.php:77]
...
#29 Magento\Framework\App\Http\Interceptor->launch() called at [lib/internal/Magento/Framework/App/Bootstrap.php:263]
#30 Magento\Framework\App\Bootstrap->run() called at [pub/index.php:29]
</pre>

Exception #1 (Magento\Framework\Exception\FileSystemException): The "/sites/mage24developee/pub/media/customer/tmp/adobe-base.jpg" file doesn't exist.
<pre>#1 Magento\Framework\Filesystem\Directory\Write->renameFile() called at [app/code/Magento/Customer/Model/FileProcessor.php:232]
#2 Magento\Customer\Model\FileProcessor->moveTemporaryFile() called at [app/code/Magento/Customer/Model/Metadata/Form/File.php:313]
#3 Magento\Customer\Model\Metadata\Form\File->processUiComponentValue() called at [app/code/Magento/Customer/Model/Metadata/Form/File.php:297]
...
#30 Magento\Framework\App\Http->launch() called at [generated/code/Magento/Framework/App/Http/Interceptor.php:23]
#31 Magento\Framework\App\Http\Interceptor->launch() called at [lib/internal/Magento/Framework/App/Bootstrap.php:263]
#32 Magento\Framework\App\Bootstrap->run() called at [pub/index.php:29]
</pre>
```
**AFTER** switching to the PR
:x: "New Customer Attribute " is not a valid URL. message appears
not_valid_url
The Customer is created successfully if the URL is edited
On the Storefront a _Please enter a valid URL. Protocol is required (http://, https:// or ftp://)._ message appears before the customer is created
store

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-30963`.

**Pull request merged**